### PR TITLE
Jira/hs 91 scanning

### PR DESCRIPTION
**Pull request by mfrazier63**
## JIRA Link
https://issues.opennms.org/browse/HS-91
https://issues.opennms.org/browse/HS-56

## Checklist
* [x] JIRA has been updated to 'in review'.
* [x] Unit tests have been added as part of the PR.
* [x] Manual tests have been completed, and notes added to JIRA.
* [ ] Details have been provided to docs team (if necessary)
* [ ] Screenshot(s) included (applicable to changes with a visual impact).
* [x] Fix comment in persistence.xml
* [x] Change index name in persistence.xml

## Details of the change

This now includes the initial provision service work done under HS-56

Additionally, this adds in the the ability to initiate a scan of nodes through a REST service call. It provides basic scanning functionality over gRPC by reading all requisitions, iterating over them, and calling out to the minions to run the detectors.

### CODE REVIEW

**Review by artnaseef — APPROVED** [review]

File: platform/provision/persistence/src/test/java/org/opennms/netmgt/provision/persistence/dto/RequisitionDTOTest.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import static org.junit.Assert.*;
+
+import com.google.gson.Gson;
+import java.net.InetAddress;
+import java.util.Date;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Test;
+import org.opennms.horizon.core.lib.InetAddressUtils;
+
```
[the remark this answers is not in the document]
Comment (mfrazier63) [reply platform/provision/persistence/src/test/java/org/opennms/netmgt/provision/persistence/dto/RequisitionDTOTest.java:11]: It's literally just a easy way to generate JSON test data. I'll remove it.

File: platform/provision/persistence/src/main/java/org/opennms/netmgt/provision/persistence/dto/RequisitionMetaDataDTO.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import lombok.Data;
+
+@Data
+public class RequisitionMetaDataDTO {
+
+    protected String context;
+    protected String key;
+    protected String value;
+
+//    @Override
```
[the remark this answers is not in the document]
Comment (mfrazier63) [reply platform/provision/persistence/src/main/java/org/opennms/netmgt/provision/persistence/dto/RequisitionMetaDataDTO.java]: Update Lombok annotations for generation of those classes, and removed the commented out impls.

File: platform/provision/persistence/src/test/java/org/opennms/netmgt/provision/persistence/dto/RequisitionDTOTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mfrazier63) [reply platform/provision/persistence/src/test/java/org/opennms/netmgt/provision/persistence/dto/RequisitionDTOTest.java]: It's literally just a easy way to generate JSON test data. I'll remove it.

File: platform/db/dao-impl/src/main/java/org/opennms/horizon/db/dao/util/AbstractDaoHibernate.java [hunk 3]
```diff
@@ -1,99 +1,93 @@
-package org.opennms.horizon.db.dao.impl;
+package org.opennms.horizon.db.dao.util;
```
Comment (geraldhumphries) [inline platform/db/dao-impl/src/main/java/org/opennms/horizon/db/dao/util/AbstractDaoHibernate.java:1]: I think this should stay in the same package as its subclasses.

File: platform/db/schema-liquibase/src/main/liquibase/provision/requisitions.xml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+		</preConditions> 
+
+		<createTable tableName="requisitions">
+
+			<!-- Unique identifier -->
+			<column name="requisitionname" type="text">
+				<constraints nullable="false" primaryKey="true" primaryKeyName="pk_requisitionname" />
+			</column>
+
+			<!-- A reference to the eventUei that created this alarm. -->
```
Comment (geraldhumphries) [inline platform/db/schema-liquibase/src/main/liquibase/provision/requisitions.xml:22]: nit: incorrect comment

File: platform/db/schema-liquibase/src/main/liquibase/provision/requisitions.xml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+	xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext"
+	xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-2.0.xsd
+		http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd">
+
+	<changeSet author="mfrazier" id="xx.xx.xx-requisitions">
+		<preConditions onFail="MARK_RAN">
+			<not><tableExists tableName="requisitions" /></not>
+		</preConditions> 
+
+		<createTable tableName="requisitions">
```
Comment (geraldhumphries) [inline platform/db/schema-liquibase/src/main/liquibase/provision/requisitions.xml:15]: Something to discuss: single or plural table names? Horizon Classic uses a mix. I will document my suggestions as part of https://issues.opennms.org/browse/HS-137.

File: platform/db/schema-liquibase/src/main/liquibase/changelog.xml [hunk 6]
```diff
@@ -95,6 +95,7 @@
 	<include file="27.1.0/changelog.xml"/>
 	<include file="29.0.0/changelog.xml"/>
 	<include file="29.0.4/changelog.xml"/>
+	<include file="provision/requisitions.xml"/>
```
Comment (geraldhumphries) [inline platform/db/schema-liquibase/src/main/liquibase/changelog.xml:98]: We still need to decide our Liquibase workflow. This is fine for now, but we will need to discuss.

File: platform/db/dao-impl/src/main/java/org/opennms/horizon/db/dao/util/AbstractDaoHibernate.java (hunk 3 above)
Comment (mfrazier63) [reply platform/db/dao-impl/src/main/java/org/opennms/horizon/db/dao/util/AbstractDaoHibernate.java:1]: Unfortunately, packages named "imp" are not exported in OSGi land, hence the move.

File: platform/db/schema-liquibase/src/main/liquibase/provision/requisitions.xml (hunk 4 above)
Comment (mfrazier63) [reply platform/db/schema-liquibase/src/main/liquibase/provision/requisitions.xml:22]: Fixed.

**Review by geraldhumphries — APPROVED** [review]

**Review by geraldhumphries — APPROVED** [review]

**Pull request merged**